src/nifreeze/viz/motion_viz.py

```diff
@@ -0,0 +1,267 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+
+from typing import Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.axes import Axes
+from scipy.ndimage import gaussian_filter
+
+
+def _get_axis(orientation: str):
+    """Get axis for the given orientation.
+
+    Parameters
+    ----------
+    orientation : :obj:`str`
+        Orientation. Cn be one of ``axial``, ``coronal``, or ``sagittal``.
+
+    Returns
+    -------
+    :obj:`int`
+        Axis corresponding to the given orientation.
+    """
+
+    if orientation == "axial":
+        return 2
+    elif orientation == "coronal":
+        return 1
+    elif orientation == "sagittal":
+        return 0
+    else:
+        raise ValueError(
+            f"Orientation must be one of 'axial', 'coronal', 'sagittal'. Provided: {orientation}"
+        )
+
+
+def _extract_slice(img_data: np.ndarray, orientation: str, slice_idx: int):
+    """Extract slice data from the given volume at the given orientation slice index.
+
+    Parameters
+    ----------
+    img_data : :obj:`~numpy.ndarray`
+        Image data to be sliced.
+    orientation : :obj:`str`
+        Orientation.
+    slice_idx : :obj:`int`
+        Slice index.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Image slice.
+    """
+
+    axis = _get_axis(orientation)
```

Comment:
```suggestion
    axis = ORIENTATIONS.index(orientation)
```

where orientations is a list of strings.